scripts/release/orthoinference/build_orthoinference_jar.pl

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+use autodie qw/:all/;
+use Cwd;
+use File::Basename;
+use File::Spec;
+use Getopt::Long;
+
+use GKB::Config;
+use GKB::CommonUtils;
+
+my ($user, $password, $database, $host, $port, $orthopairs_path, $release, $release_date, $person_id);
+GetOptions(
+    'user:s' => \$user,
+    'pass:s' => \$password,
+    'db:s' => \$database,
+    'host:s' => \$host,
+    'port:i' => \$port,
+    'orthopairs_path:s' => \$orthopairs_path,
+    'release:s' => \$release,
+    'release_date:s' => \$release_date,
+    'person_id:s' => \$person_id,
+);
+$release || die "Need Reactome release version: -release [release version]\n";
+$release_date || die "Need release date: -release_date [yyyy-mm-dd]\n";
+$person_id || die "Need person id: -person_id [db id of person instance for orthoinference script]\n";
+$user ||= $GKB::Config::GK_DB_USER;
+$password ||= $GKB::Config::GK_DB_PASS;
+$database ||= $GKB::Config::GK_DB_NAME;
+$host ||= $GKB::Config::GK_DB_HOST;
+$port ||= $GKB::Config::GK_DB_PORT;
+$orthopairs_path ||= File::Spec->catfile(dirname(getcwd), 'orthopairs', $release, '/');
+
+my $starting_directory = getcwd;
+my $orthoinference_repository = 'data-release-pipeline';
+
+if (! (-d "$orthoinference_repository/.git")) {
+    system "git clone https://github.com/reactome/$orthoinference_repository.git";
+}
+chdir $orthoinference_repository;
+system 'git pull';
+run_command('git checkout develop', {
+    ignore_error => qr/^Already on .*/
+});
+chdir 'release-common-lib';
+system 'mvn clean install';
+my $orthoinference_project_dir = File::Spec->catfile($starting_directory, $orthoinference_repository, 'orthoinference');
+chdir $orthoinference_project_dir;
+
+my $resources_dir = File::Spec->catfile($orthoinference_project_dir, 'src', 'main', 'resources');
+my $config_file = File::Spec->catfile($resources_dir, 'config.properties');
+my $species_config = File::Spec->catfile($resources_dir, 'Species.json');
+create_config_properties_file($config_file, {
+    username => $user,
+    password => $password,
+    database => $database,
+    host => $host,
+    port => $port,
+    pathToSpeciesConfig => $species_config,
+    pathToOrthopairs => $orthopairs_path,
+    releaseNumber => $release,
+    dateOfRelease => $release_date,
+    personId => $person_id,
+});
+system "git update-index --assume-unchanged $config_file"; 
+system "ln -sf $starting_directory/normal_event_skip_list.txt $resources_dir";
+system "ln -sf $orthoinference_project_dir/runOrthoinference.sh $starting_directory/runOrthoinference.sh";
+
+sub create_config_properties_file {
+    my $config_properties_file = shift;
+    my $options = shift;
+    open(my $config_fh, '>', $config_properties_file);
+    my @properties = map { $_ . '=' . $options->{$_} } keys %{$options};
+    foreach my $property (@properties) {
+        print {$config_fh} "$property\n";
+    }
+    close $config_fh;
+
+    return;
+}
```

Comment:
Maybe I missed something here, but I read all the way to the end and I didn't see anything that _builds_ orthoinference. Only a command that builds the release-common-lib.

Reply:
Yeah, I think this script is misnamed.  Maybe something like 'pre_orthoinference_setup.pl'

Reply:
A brand-new script that's _already_ deprecated, before merging back?

Reply:
So... is the code that actually builds orthoinference in this PR?

Reply:
No, run_orthoinference.sh builds the jar file.  This just does some setup of sym-links, configuration files, and the release-common-lib build.

Reply:
Why is it in two separate scripts? Why not one?

Reply:
It could be combined I suppose.  run_orthoinference.sh was a data-release-pipeline repository script where this is a Release repository script.  Setting up the configuration file is a bit easier in Perl because it's easier to parsed named arguments, but I'm sure it's doable in BASH.  I just haven't done it before.

Reply:
Hmm ok. Well, I'd suggest moving the release-common-lib build to the other shell script since it seems weird to build one dependency in a Perl script and then build the application in a shell script. 

BUUUUUT.... since release-common-lib is now in artifactory, maybe @cookersjs can modify Orthoinference to get the dependency directly from artifactory, and then that chunk of code can be removed from _this_ script and it will make more sense to rename it to something like `setup_orthoinference`, or something like that... 

So for now... maybe just rename?

Reply:
That sounds good.  I'll rename and before I merge this PR, we'll update orthoinference's POM.  Then I'll remove the release-common-lib build.

Reply:
I'll add this as an issue for the repo